src/core/font_substitutions.js

```diff
@@ -456,19 +456,29 @@ function generateFont(
  * @param {String} baseFontName The font name to be substituted.
  * @param {String|undefined} standardFontName The standard font name to use
  *   if the base font is not available.
+ * @param {String} type The font type.
  * @returns an Object with the CSS, the loaded name, the src and the style.
  */
 function getFontSubstitution(
   systemFontCache,
   idFactory,
   localFontPath,
   baseFontName,
-  standardFontName
+  standardFontName,
+  type
 ) {
   if (baseFontName.startsWith("InvalidPDFjsFont_")) {
     return null;
   }
 
+  if (
+    (type === "TrueType" || type === "Type1") &&
+    /^[A-Z]{6}\+/.test(baseFontName)
+  ) {
+    // When the font is a subset, we need to remove the prefix (see 9.6.4).
+    baseFontName = baseFontName.slice(7);
```

Comment:
Could you add the PDF file from the issue as a reference test so we have a PDF file that exercises this new logic to avoid regressions?